azurerm/helpers/azure/resourceid.go

```diff
@@ -54,7 +54,8 @@ func ParseAzureResourceID(id string) (*ResourceID, error) {
 
 		// Catch the subscriptionID before it can be overwritten by another "subscriptions"
 		// value in the ID which is the case for the Service Bus subscription resource
-		if key == "subscriptions" && subscriptionID == "" {
+		// Also handle different casing of Subscription that happens when talking to recovery services API
```

Comment:
Given this is a unique bug with the recovery services API we should handle it in the resources with a comment alluding to why, and a link to this open SDK issue: https://github.com/Azure/azure-sdk-for-go/issues/2824